README.md

```diff
@@ -76,4 +76,80 @@ $ singularity build osgvo-pilot.sif docker://opensciencegrid/osgvo-docker-pilot
 ```
 
 
+## CVMFS Without a Bind-Mount Using cvmfsexec
+
+If you don't have CVMFS available on the host, the container can still make
+CVMFS available by using [cvmfsexec](https://github.com/cvmfs/cvmfsexec#readme).
+
+This will require a kernel version >= 3.10.0-1127 on an EL7-compatible host
+or >= 4.18 on an EL8-compatible host, or user namespaces enabled by hand --
+see the cvmfsexec README linked above for details.
+
+This will also require granting the container some additional privileges, which
+you can do in one of two ways:
+
+1.  Add `--privileged` to the `docker run` invocation.
+
+2.  Add
+    `--security-opt seccomp=unconfined --security-opt systempaths=unconfined --device=/dev/fuse`
+    to the `docker run` invocation.
+
+The second option will add only the minimum necessary privileges for cvmfsexec.
+
+Using cvmfsexec takes place in the entrypoint, which means it will still happen
+even if you specify a different command to run, such as `bash`.  You can bypass
+the entrypoint by passing `--entrypoint <cmd>` where `<cmd>` is some different
+command to run, e.g. `--entrypoint bash`.  Setting the entrypoint this way
+clears the command.
+
+There are several environment variables you can set for cvmfsexec:
+
+-   `NO_CVMFSEXEC` - set this to disable cvmfsexec.
+
+-   `CVMFSEXEC_REPOS` - this is a comma-separated list of CVMFS repos to mount, if using cvmfsexec.
+    Set this to blank to avoid using cvmfsexec, in which case you won't need to
+    add the above permissions to your container.  The default is to mount the
+    OASIS repo (oasis.opensciencegrid.org).
+
+-   `CVMFS_HTTP_PROXY` - this sets the proxy to use for CVMFS; leave this blank
+    to find the best one via wlcg-lpad.
+
+-   `CVMFS_QUOTA_LIMIT` - the quota limit in MB for CVMFS; leave this blank to
+    use the system default (4 GB)
+
+
+You can store the cache outside of the container by bind-mounting a directory
+to `/cvmfs-cache`.
+You can store the logs outside of the container by bind-mounting a directory to
+`/cvmfs-logs`.
+
```

Comment:
```suggestion
You can customize cvmfs further by bind-mounting a file on
`/cvmfsexec/dist/etc/cvmfs/default.local`.
```

Reply:
Oh but now I realize this won't work because of the tarball extraction.  You would have to choose another path.  How about `/cvmfsexec/default.local` and then look for that and copy it into the final path if present.

Reply:
Sure, that would work. If you bind-mount `/cvmfsexec/default.local` but also set the environment variables such as `CVMFS_HTTP_PROXY`, which should win?

Reply:
I think you could do it whichever way is more convenient.  I would guess given your current implementation that the variables would win.